client/src/redux/selectors.js

```diff
@@ -29,26 +32,38 @@ export const showCertSelector = state => state[MainApp].showCert;
 export const showCertFetchStateSelector = state =>
   state[MainApp].showCertFetchState;
 export const shouldRequestDonationSelector = state => {
-  const completedChallenges = completedChallengesSelector(state);
+  const completedChallengesLength = completedChallengesSelector(state).length;
   const completionCount = completionCountSelector(state);
-  const canRequestProgressDonation = state[MainApp].canRequestProgressDonation;
+  const sessionChallNumProgressModalShown =
+    sessionChallNumProgressModalShownSelector(state);
   const isDonating = isDonatingSelector(state);
   const recentlyClaimedBlock = recentlyClaimedBlockSelector(state);
+  const showMultipleProgressModals = showMultipleProgressModalsSelector(state);
 
   // don't request donation if already donating
   if (isDonating) return false;
 
   // a block has been completed
   if (recentlyClaimedBlock) return true;
 
+  /* 
+  When AB testing for showing multiple progress modals is active,
+  show a donation modal every 30 challenges after the first 50
+   */
+  if (
+    showMultipleProgressModals &&
+    sessionChallNumProgressModalShown !== null &&
+    completedChallengesLength > 50 &&
+    sessionChallNumProgressModalShown + 20 >= completionCount
+  )
+    return true;
+
   // a donation has already been requested
-  if (!canRequestProgressDonation) return false;
+  if (sessionChallNumProgressModalShown !== null) return false;
```

Comment:
```suggestion
  if (progressDonationModalShown) return false;
```
This is also a little better.

Reply:
The last commit should have resolved this.